src/test/storage/chunk_test.cpp

```diff
@@ -58,4 +59,35 @@ TEST_F(StorageChunkTest, RetrieveSegment) {
   EXPECT_EQ(base_segment->size(), 4u);
 }
 
+TEST_F(StorageChunkTest, ApplyDictionaryEncoding) {
+  c.add_segment(int_value_segment);
+  c.add_segment(string_value_segment);
+  for (int i = 0; i < 10; ++i) {
+    c.append({3, "two"});
+  }
+
+  std::vector<std::string> column_types{"int", "string"};
+
+  auto encoded_chunk = c.apply_dictionary_encoding(c, column_types);
+
+  EXPECT_EQ(encoded_chunk->column_count(), 2u);
+  EXPECT_EQ(encoded_chunk->size(), 13u);
```

Comment:
Having a size of thirteen is obviously correct, but we had go back and have a look why this is correct. Maybe a more clear creation of testing segments could help readability